python/tskit/tables.py

```diff
@@ -3452,43 +3452,11 @@ def union(
 
     def find_ibd(self, samples, max_time=None, min_length=None):
         """
-        Returns a dictionary containing all IBD (identical-by-descent) segments for the
-        supplied list of sample pairs.
-
-        A pair of nodes ``(u, v)`` has an IBD segment with a left and right coordinate
-        ``[left, right)`` and ancestral node ``a`` iff the most recent common ancestor
-        of the segment ``[left, right)`` in nodes ``u`` and ``v`` is ``a``, and the
-        segment has been inherited along the same genealogical path (ie. it has not
-        been broken by recombination). The segments returned are the longest possible
-        ones.
-
-        Note that this definition is purely genealogical -- allelic states
-        *are not* considered here. If used without time or length thresholds, the
-        segments returned for a given pair will partition the span of the contig
-        represented by the tree sequence.
-
-        Each item in the outputted dictionary object contains the IBD segments for a
-        particular sample pair. Suppose the output is
-
-        .. code-block:: python
-
-            {
-                (0, 1): {
-                    "left": array([0.0]),
-                    "right": array([1.0]),
-                    "node": array([4]),
-                },
-                (0, 2): {
-                    "left": array([0.5, 0.0]),
-                    "right": array([1.0, 0.5]),
-                    "node": array([3, 5]),
-                },
-            }
-
-        This indicates that samples 0 and 1 share a single IBD segment on the interval
-        `[0.0, 1.0)` inherited from the ancestor with node ID 4. Samples 0 and 2 share
-        one IBD segment on the interval `[0.5, 1.0)` inherited from node 3, and another
-        segment on the interval `[0.0, 0.5)` inherited from node 5.
+        Same as :meth:`TreeSequence.find_ibd`. See this method for more info.
+        The table collection must correspond to a valid :class:`TreeSequence` and must
+        be sorted. Use :meth:`TableCollection.sort` beforehand to ensure this. To ensure
+        that any adjacent IBD segments are concatenated into a single segment in the
```

Comment:
Some will read this as telling them to *always* run ``squash``, but this is only needed in special cases right? (Which are quite remote, I think?)

Reply:
I think, now that we've made the 'default' method the `TreeSequence` one, people are only going to use the TableCollection version of this function directly when they've manually constructed the tables themselves. In this case, I think I'd be encouraging them to squash and sort the tables as best practice anyway, because they have probably been deliberately constructed to be special in some way.  Perhaps I could add a comment recommending that people use the TreeSequence version of this method unless they specifically need this one?

Reply:
That's a good idea. Sorting by default wouldn't be a good practise to encourage though, as this is very expensive for large tables.